aioreactive/transform.py

```diff
@@ -465,3 +466,71 @@ def factory(exn: Exception) -> AsyncObservable[_TSource]:
         return pipe(source, catch(factory))
 
     return _retry
+
+
+def _scan(
+    accumulator: Callable[[_TResult, _TSource], Union[Awaitable[_TResult], _TResult]],
```

Comment:
I think I would like the accumulator to always be awaitable to avoid all the checking inside the function:
```suggestion
    accumulator: Callable[[_TResult, _TSource], Awaitable[_TResult]],
```